src/EditorFeatures/Core/Implementation/Formatting/FormatCommandHandler.cs

```diff
@@ -121,32 +134,49 @@ public void ExecuteReturnOrTypeCommand(EditorCommandArgs args, Action nextHandle
                 return;
             }
 
-            var service = document.GetLanguageService<IEditorFormattingService>();
+            var service = document.GetLanguageService<IFormattingInteractionService>();
+            IEditorFormattingService? editorService = null;
             if (service == null)
             {
-                return;
+                editorService = document.GetLanguageService<IEditorFormattingService>();
+                if (editorService == null)
+                {
+                    return;
+                }
             }
```

Comment:
so this is technically correct.  however consider a simpler form of:

```
var service1 = ...
var service2 = ...
if (service1 is null && service2 is null)
    return;
```

It jsut reads so much more clearly :)